`@plone/registry` as ESM module, move to TS, complete documentation (#6399)
plone/volto: packages/registry/docs/howto/how-to-register-an-addon.md

- `VOLTOCONFIG`

The value of the environment variable must point to a file that exists relative to the app folder (the one you pass to the instatiation of the Add-on Registry).
You can pass also the full path of the file.

For example, if your configuration file is named {file}`my-add-on-registry-config.js` and is located at the root of your add-on package, you would set your environment variable as shown.

```shell
set REGISTRYCONFIG="my-add-on-registry.config.js"
```

If the file that you specify in the environment variable exists, then `@plone/registry` uses it to configure your add-on.
If it does not exist, then `@plone/registry` looks for the configuration file in the following locations in the root of your app in order.
The first found configuration file wins.

- `registry.config.js`
- `volto.config.js`

This is an example of a configuration file.
You must define it in [CommonJS](https://en.wikipedia.org/wiki/CommonJS) format.

```js
module.exports = {
  addons: ['my-volto-config-addon'],
};
```

If your app is in ESM (`"type": "module"` in `package.json`), then you should use the `.cjs` suffix for the configuration file to mark it as a proper `CommonJS` file.

> If `@plone/registry` finds no configuration file, then it only relies on the configuration (if any) in the `addons` key in `package.json`.

Author: I added a couple of phrases at the top and here.

# How to register an add-on in your app

You have two ways available for registering an add-on in your app.

> Via `package.json` key or using a configuration file.

Reviewer: ```suggestion
You can do so either through the `addons` key in your {file}`package.json` or by using a configuration file.
```

## Via `addons` key in `package.json`

> 

Reviewer: ```suggestion

The following code sample shows how to register your add-on in your app through the `addons` key in your {file}`package.json`.

```

  return config
};
```

## Via a configuration file

The configuration file can be provided via an environment variable.

> Because of legacy reasons, you can use one of these two:

Reviewer: Is one env var preferred over the other? If so, we should state such.

```suggestion
You can use one of these two environment variables.
```

> - `REGISTRYCONFIG`.

Reviewer: ```suggestion
- `REGISTRYCONFIG`
```

- `VOLTOCONFIG`

> 

Reviewer: Fill in the blank, and adjust text and code example.

````suggestion

The value of the environment variable must point to a file that exists relative to the ______.

For example, if your configuration file is named {file}`my-add-on-config.json` and is located at the root of your add-on package, you would set your environment variable as shown.

```shell
set REGISTRYCONFIG="my-add-on-config.json"
```

````

> If it exists, `@plone/registry` uses the file provided as the add-ons configuration.

Reviewer: ```suggestion
If the file that you specify in the environment variable exists, then `@plone/registry` uses it to configure your add-on.
```

> If it does not, `@plone/registry` looks up for the configuration file existing in these places:

Reviewer: Please clean up this suggestion to be accurate.

```suggestion
If it does not exist, then `@plone/registry` looks for the configuration file in the following locations in the root of your app in order.
The first found configuration file wins.
If `@plone/registry` finds no configuration file, then it 💩 the bed.  [TODO: What does it actually do?]
```

- `registry.config.js`
- `volto.config.js`

> in the root of your app, in this order.

Reviewer: ```suggestion
```

> If (one of them) exists, it will load it.

Reviewer: ```suggestion
```

This is an example of a configuration file.

> 

Reviewer: ```suggestion
You must define it in [CommonJS](https://en.wikipedia.org/wiki/CommonJS) format.

```

```js
module.exports = {
  addons: ['my-volto-config-addon'],
};
```

> It has to be defined in commonJS format.

Reviewer: ```suggestion
```
